core/src/main/scala/scalacache/serdes/JavaSerializationCodec.scala

```diff
@@ -0,0 +1,72 @@
+package scalacache.serdes
+
+import java.io._
+
+import scala.reflect.ClassTag
+import scala.util.control.NonFatal
+
+trait JavaSerializationCodec {
+
+  private[this] class GenericCodec[S <: Serializable](classTag: ClassTag[S]) extends Codec[S] {
+
+    def using[T <: Closeable, R](obj: T)(f: T => R): R =
+      try
+        f(obj)
+      finally
+        try obj.close() catch {
+          case NonFatal(_) => // does nothing
+        }
+
+    def serialize(value: S): Array[Byte] =
+      using (new ByteArrayOutputStream()) { buf =>
+        using (new ObjectOutputStream(buf)) { out =>
+          out.writeObject(value)
+          out.close()
+          buf.toByteArray
+        }
+      }
+
+    def deserialize(data: Array[Byte]): S =
+      using (new ByteArrayInputStream(data)) { buf =>
+        val in = new GenericCodecObjectInputStream(classTag, buf)
+        using (in) { inp =>
+          inp.readObject().asInstanceOf[S]
+        }
+      }
+  }
+
+  /**
+    * Uses plain Java serialization to deserialize objects
+    */
+  implicit def AnyRefBinaryCodec[S <: Serializable](implicit ev: ClassTag[S]): Codec[S] =
+    new GenericCodec[S](ev)
+
+}
+
+/**
+  * Object input stream which tries the thread local class loader.
+  *
+  * Thread Local class loader is used by SBT to avoid polluting system class loader when
+  * running different tasks.
+  *
+  * This allows deserialization of classes from sub-projects during something like
+  * Play's test/run modes.
+  */
```

Comment:
Interesting. Maybe this could have solved #32. Or, maybe it will cause a regression in the fix for that bug...

The fix was to allow users to supply their own classloader to `RedisCache`, but that will now be ignored.

Needs a bit more investigation.